dvc/repo/remove.py

```diff
@@ -1,14 +1,22 @@
+import logging
+
 from . import locked
+from ..utils import parse_target
+
+logger = logging.getLogger(__name__)
 
 
 @locked
 def remove(self, target, outs_only=False):
     from ..dvcfile import Dvcfile
 
-    stage = Dvcfile(self, target).stage
-    if outs_only:
+    path, name, _ = parse_target(target)
+    dvcfile = Dvcfile(self, path)
+    stages = list(dvcfile.stages.filter(name).values())
+    for stage in stages:
         stage.remove_outs(force=True)
-    else:
-        stage.remove(force=True)
 
-    return stage
+    if not outs_only:
+        dvcfile.remove()
```

Comment:
if we will remove one particular stage, the whole dvcfile will be removed? This seems odd.

Reply:
It's handled inside `dvcfile.remove()` here. For single stage file, it's perfectly okay to delete it, but multistage file will deny removing the file unless forced. In short, pipeline file won't get deleted, single stage will be.